### ISSUE TITLE
Speed up mget and other similar calls, by applying the pipeline parrelllel exection speedup under the hood. 

### ISSUE THREAD

**Issue by AlexBoyd** [Grokzen/redis-py-cluster#488]
As far as I can tell, the current implementation for commands like mget is to unroll them into a sequential series of get calls, to ensure the correctness, but that leads to very slow performance for calls for a large quantity of keys. See (https://github.com/Grokzen/redis-py-cluster/issues/375) 

Would it be possible to apply the parallel execution speed ups that are used for pipelines though, so that the unrolled series of gets only need to block once on all of the calls at once, rather than sequentially waiting on many round trips to 1 node at a time?

**Comment by Grokzen** [comment]
@AlexBoyd I am closing this issue as this repo is no Archived and all future work and features should be redirected over to redis-py instead
